### PR TITLE
OboeTester: add error callback test

### PR DESCRIPTION
**Pull request by philburk**
Delete an error callback while it is being used by Oboe.

Try to reproduce issue #1603

### CODE REVIEW

File: apps/OboeTester/app/src/main/cpp/TestErrorCallback.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            usleep(10 * 1000); // Force us to lose the race condition.
+            // The callback was probably deleted by now.
+            LOGE("%s() - error = %s, mMagic = 0x%08X",
+                 __func__, oboe::convertToText(error), mMagic.load());
+            mParent->mCallbackMagic = mMagic.load();
+        }
+
+    private:
+        TestErrorCallback *mParent;
+        static constexpr int32_t kMagicGood = 0x2B4ACAFE;
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/cpp/TestErrorCallback.h:95]: Can we use something like 0x600DCAFE? It's hard to determine that 0x2B4ACAFE is the intended good value

File: apps/OboeTester/app/src/main/cpp/TestErrorCallback.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            // reported in #1603.
+            std::thread t([this]() {
+                    this->mParent->cleanup();
+                    LOGE("onErrorBeforeClose called cleanup!");
+                });
+            t.detach();
+        }
+
+        void onErrorAfterClose(oboe::AudioStream *oboeStream, oboe::Result error) override {
+            usleep(10 * 1000); // Force us to lose the race condition.
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/cpp/TestErrorCallback.h:86]: I'm confused why we need the sleep here. What is the order of events for the bad and good cases?

File: apps/OboeTester/app/src/main/cpp/TestErrorCallback.h (hunk 1 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/cpp/TestErrorCallback.h:95]: Nice! will do.

File: apps/OboeTester/app/src/main/cpp/TestErrorCallback.h (hunk 2 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/cpp/TestErrorCallback.h:86]: We want to try to delete the callback before onErrorAfterClose() is run.
It is a race condition and I want the thread doing the deletion to win the race.
After we get it to FAIL reliably then I will fix it by adding an API that uses a shared_ptr.

**Pull request merged**